cli/generators/crud/templates/components/addPage.js

```diff
@@ -0,0 +1,72 @@
+import React, { PropTypes } from 'react';
+import { browserHistory } from 'react-router';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import toastr from 'toastr';
+import * as <%= name %>Actions from '../../actions/<%= name %>Actions';
+import <%= ucName %>Form from './<%= ucName %>Form';
+
+class <%= ucName %>AddPage extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      <%= name %>: Object.assign({}, props.<%= name %>),
+      saving: false
+    };
+
+    this.handleSave = this.handleSave.bind(this);
+  }
+
+  handleSave(<%= name %>) {
+    this.setState({ saving: true });
+    this.props.actions.create<%= ucName %>(<%= name %>)
+      .then(() => {
+        this.setState({ saving: false });
+        toastr.success('<%= ucName %> created successfully');
+        browserHistory.push('/app/<%= pluralizedName %>');
+      })
+      .catch(error => {
+        this.setState({ saving: false });
```

Comment:
Hehehe that one is the reason I was looking for a `finally` statement the other day.

Yep, it works... but... we want to do something like that? Asking because it could not be as intuitive as the current approach. Which approach do you prefer folks? Current one or the suggested by @AlphaGit where the last "then" works like a "finally"?

@AlphaGit @juampick @mravinale